cli/pkg/kctrl/cmd/app/release/release.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package release
+
+import (
+	"os"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/spf13/cobra"
+	cmdappbuild "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/app/init"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/logger"
+)
+
+type ReleaseOptions struct {
+	ui          cmdcore.AuthoringUI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	pkgVersion     string
+	chdir          string
+	outputLocation string
+	debug          bool
+}
+
+const (
+	defaultArtifactDir = "carvel-arteficts"
+	defaultVersion     = "0.0.0-%d"
+)
+
+func NewReleaseOptions(ui ui.UI, depsFactory cmdcore.DepsFactory, logger logger.Logger) *ReleaseOptions {
+	return &ReleaseOptions{ui: cmdcore.NewAuthoringUIImpl(ui), depsFactory: depsFactory, logger: logger}
+}
+
+func NewReleaseCmd(o *ReleaseOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "release",
+		Short: "Release package",
+		RunE:  func(cmd *cobra.Command, args []string) error { return o.Run() },
+	}
+
+	cmd.Flags().StringVarP(&o.pkgVersion, "version", "v", "", "Version to be released")
+	cmd.Flags().StringVar(&o.chdir, "chdir", "", "Working directory with package-build and other config")
+	cmd.Flags().StringVar(&o.outputLocation, "copy-to", defaultArtifactDir, "Output location for artifacts")
+	cmd.Flags().BoolVar(&o.debug, "debug", false, "Version to be released")
+
+	return cmd
+}
+
+func (o *ReleaseOptions) Run() error {
+	if o.chdir != "" {
+		err := os.Chdir(o.chdir)
```

Comment:
Reconciller fetch has an issue if package-build.yml is not in the root dir.

Reply:
Putting in here from discussion we had.

Chdir is not to be used in most cases, and when used it is reasonable to expect config files to reside in the folder itself.
Since, our primary intention here was to move the files created by us to a folder that is not the root, we should be doing that by default if that seems to be preferred.